src/backend/distributed/transaction/backend_data.c

```diff
@@ -945,3 +960,68 @@ GetMyProcLocalTransactionId(void)
 {
 	return MyProc->lxid;
 }
+
+
+/*
+ * get_all_active_client_backend_count returns the active
+ * client backend count.
+ */
+Datum
+get_all_active_client_backend_count(PG_FUNCTION_ARGS)
+{
+	PG_RETURN_UINT32(GetAllActiveClientBackendCount());
+}
+
+
+/*
+ * GetAllActiveClientBackendCount returns activeClientBackendCounter in
+ * the shared memory.
+ */
+int
+GetAllActiveClientBackendCount(void)
+{
+	uint32 activeBackendCount =
+		pg_atomic_read_u32(&backendManagementShmemData->activeClientBackendCounter);
+
+	return activeBackendCount;
+}
+
+
+/*
+ * IncrementClientBackendCounter increments activeClientBackendCounter in
+ * the shared memory by one.
+ */
+void
+IncrementClientBackendCounter(void)
+{
+	if (GetMaxSharedPoolSize() == DISABLE_CONNECTION_THROTTLING)
+	{
+		/*
+		 * We only care about activeClientBackendCounter when adaptive
+		 * connection management is enabled.
+		 */
+		return;
+	}
+
+	pg_atomic_add_fetch_u32(&backendManagementShmemData->activeClientBackendCounter, 1);
+}
+
+
+/*
+ * DecrementClientBackendCounter decrements activeClientBackendCounter in
+ * the shared memory by one.
+ */
+void
+DecrementClientBackendCounter(void)
+{
+	if (GetMaxSharedPoolSize() == DISABLE_CONNECTION_THROTTLING)
+	{
+		/*
+		 * We only care about activeClientBackendCounter when adaptive
+		 * connection management is enabled.
+		 */
```

Comment:
To me it seems a bit nicer to separate those concerns and always count backends

Reply:
I don't feel strongly about this, but I feel like this `if` is not super useful. Other than that, this PR seems fine.

Reply:
If user enables/disables `citus.max_shared_pool_size`, we'd ruin the backend counter, and won't be able to recover without a restart. So better to remove anyway, done.